### PR TITLE
added method DiscoverSQLMigrationsFromFilesystem

### PR DESCRIPTION
**Pull request by smcdonald45**
In our project we are using [vfsgen](https://github.com/shurcooL/vfsgen) for embed static content like go templates and database migrations. Unfortunately go-pg migrations only supports the os filesystem and the files have to exist on production systems. We took a look around for other migration frameworks but decided to add the vfs feature to go-pg because its already running and we also like to work with the orm.

Changes:
 - added method DiscoverSQLMigrationsFromFilesystem(fs http.FileSystem, dir string) which we can use with a filesystem generated by vfsgen
 - the file contents also loaded using the given fs instead of os.open

In the testcase you can see some generated code by vfsgen which is used for discover the SQL.

Thanks in advance,
 Stephan

### CODE REVIEW

**Comment by Terminator637** [comment]
@vmihailenco, @smcdonald45, I think this could fix my problem #73, who can merge it?

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+*.iml
```
Comment (vmihailenco) [inline .gitignore]: Can you please remove this file? Such things should go into global .gitignore file

File: collection.go [hunk 2]
```diff
@@ -170,10 +187,11 @@ func (c *Collection) DiscoverSQLMigrations(dir string) error {
 		return ms[len(ms)-1]
 	}
 
-	files, err := ioutil.ReadDir(dir)
+	files, err := f.Readdir(-1)
 	if err != nil {
 		return err
 	}
+	sort.Slice(files, func(i, j int) bool { return files[i].Name() < files[j].Name() })
```
Comment (vmihailenco) [inline collection.go]: Do we really need this? I think this package already must sort migrations as they are added.

**Comment by vmihailenco** [comment]
Thanks - this looks good. I will try to merge this tomorrow. Sorry for delay.

**Pull request merged**